### PR TITLE
Update export feature

### PR DESCRIPTION
**Pull request by AK46rocks**
> **Warning**
>
> If that is your first contribution to Metabase, please sign the [Contributor License Agreement](https://docs.google.com/a/metabase.com/forms/d/1oV38o7b9ONFSwuzwmERRMi9SYrhYeOrkbmNaq9pOJ_E/viewform) (unless it's a tiny documentation change). Also, if you're attempting to fix a translation issue, please submit your changes to our [POEditor project](https://poeditor.com/join/project/ynjQmwSsGh) instead of opening a PR.

Closes https://github.com/metabase/metabase/issues/[issue_number]

### Description

Describe the overall approach and the problem being solved.

### How to verify

Describe the steps to verify that the changes are working as expected.

1. New question -> Sample Dataset -> ...
2. ...

### Demo

_Upload a demo video or before/after screenshots if sensible or remove the section_

### Update Download Feature
- Instead of Directly Calling Analyzer API to export charts in CSV,JSON,XLSX format,  Made request from backed to call these Analyzer APIs internally
- Used for Tracking Downloads
- Added route for backend APi in `security.clj`
- Major changes in `frontend/src/metabase/components/DownloadButton.jsx` file

### CODE REVIEW

**Review by deepakkashyap3013 — APPROVED** [review]

File: frontend/src/metabase/components/DownloadButton.jsx [hunk 1]
```diff
@@ -31,19 +33,34 @@ const retrieveFilename = ({ res, type }) => {
   const contentDispositionHeader = res.headers.get("Content-Disposition") || "";
   const contentDisposition = decodeURIComponent(contentDispositionHeader);
   const match = contentDisposition.match(/filename="(?<fileName>.+)"/);
-  const fileName =
-    match?.groups?.fileName ||
-    `query_result_${new Date().toISOString()}.${type}`;
+  const fileName = match?.groups?.fileName;
+  // || `query_result_${new Date().toISOString()}.${type}`;
```
Comment (realchoubey) [inline frontend/src/metabase/components/DownloadButton.jsx:37]: Add a line of comment to explain why this commented code is not removed

File: frontend/src/metabase/components/DownloadButton.jsx [hunk 2]
```diff
@@ -55,20 +72,53 @@ const handleSubmit = async (
...
+
+      // This try-catch is used only to catch newResponse.json() error when getting response as Blob data
+      try {
+        const newResponse = new Response(blobData);
+        const downloadResponse = (await newResponse?.json()) || undefined;
+        if (
+          downloadResponse?.success === false &&
+          checkApiIsFromDappLooker(requestUrl)
+        ) {
+          // const keys = Object.keys(downloadResponse.errorData);
```
Comment (realchoubey) [inline frontend/src/metabase/components/DownloadButton.jsx:104]: Add a line of comment to explain why this commented code is not removed

File: frontend/src/metabase/components/DownloadButton.jsx [hunk 3]
```diff
@@ -55,20 +72,53 @@ const handleSubmit = async (
...
+          // const errorMsg = downloadResponse.errorData[keys[0]];
+          // console.log(errorMsg);
+          onDownloadRejected();
+          return;
+        }
+      } catch (error) {
+        // console.error(error);
```
Comment (realchoubey) [inline frontend/src/metabase/components/DownloadButton.jsx:111]: Log this exception

File: frontend/src/metabase/query_builder/components/QueryDownloadWidget/utils.ts [hunk 4]
```diff
@@ -74,9 +74,11 @@ export const getDownloadButtonParams = ({
   const isSavedQuery = card?.id != null;
   if (isSavedQuery) {
     return {
-      method: "POST",
-      url: `api/card/${card.id}/query/${type}`,
+      method: "GET",
+      // url: `api/card/${card.id}/query/${type}`,
```
Comment (realchoubey) [inline frontend/src/metabase/query_builder/components/QueryDownloadWidget/utils.ts:78]: Add a line of comment to explain why this commented code is not removed

File: frontend/src/metabase/visualizations/lib/save-chart-image.ts [hunk 5]
```diff
@@ -59,9 +68,26 @@ export const saveChartImage = async (selector: string, fileName: string, isShowW
...
   link.remove();
+
+  //Save Record in Download Stats
+  const requestUrl = `https://dapplooker.com/web/chart/download?source=analyzer&exportFormat=png&cardId=${cardId}&coreUserId=${coreUserId}`;
+
+  fetch(requestUrl, { method: "GET" })
+    .then(async res => {
+      const downloadResponse = await res.json();
+      if (downloadResponse.success === false) {
+        // const keys = Object.keys(downloadResponse.errorData);
```
Comment (realchoubey) [inline frontend/src/metabase/visualizations/lib/save-chart-image.ts:86]: Add a line of comment to explain why this commented code is not removed

File: src/metabase/api/card.clj [hunk 6]
```diff
@@ -988,7 +988,7 @@ saved later when it is ready."
   (qp.card/run-query-for-card-async
    card-id export-format
    :parameters  (json/parse-string parameters keyword)
-   :constraints nil
+  ;;  :constraints nil
```
Comment (realchoubey) [inline src/metabase/api/card.clj:991]: Add a line of comment to explain why this commented code is not removed

File: src/metabase/server/middleware/security.clj [hunk 7]
```diff
@@ -47,6 +47,10 @@
 
 (def cloudfront-domain "https://d2yxqfr8upg55w.cloudfront.net")
 
+(def restful-api-endpoint-dev "http://dlooker.com:8080")
+
+(def restful-api-endpoint-prod "https://dapplooker.com")
```
Comment (realchoubey) [inline src/metabase/server/middleware/security.clj:52]: Use public IP of server or use localhost.

File: frontend/src/metabase/components/DownloadButton.jsx (hunk 3 above)
Comment (deepakkashyap3013) [reply frontend/src/metabase/components/DownloadButton.jsx]: removed the log because this catch block should be empty

**Pull request merged**